docs/developer-docs/integrations/sns/managing/making-proposals.md

```diff
@@ -42,6 +42,619 @@ For these cases, SNSs have so called 'generic proposals'. These are custom propo
+        amount_e8s : nat64;
+    };
+
+    type Subaccount = record { subaccount : vec nat8 };
+```
+
+### Putting it together
+
+```candid
+    type TransferSnsTreasuryFunds = record {
+        from_treasury : int32;
+        to_principal : opt principal;
+        to_subaccount : opt record { subaccount : vec nat8 };
+        memo : opt nat64;
+        amount_e8s : nat64;
+    };
+```
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "Lorem ipsum";
+        url = "Lorem ipsum";
+        summary = "Lorem ipsum";
+        action = opt variant {
+            TransferSnsTreasuryFunds = record {
+                from_treasury = int32;
+                to_principal = opt principal;
+                to_subaccount = opt record { subaccount = vec nat8 };
+                memo = opt nat64;
+                amount_e8s = nat64;
+               
+            };
+        };
+    };
+)'
+```
+
+## `UpgradeSnsControlledCanister` 
+
+### Relevant type signatures
+
+```candid
+    type UpgradeSnsControlledCanister : UpgradeSnsControlledCanister;
+
+    type UpgradeSnsControlledCanister = record {
+        new_canister_wasm : vec nat8;
+        mode : opt int32;
+        canister_id : opt principal;
+        canister_upgrade_arg : opt vec nat8;
+    };
+```
+
+### Putting it together
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "lorem ipsum";
+        url = "lorem ipsum";
+        summary = "lorem ipsum";
+        action = opt variant {
+            UpgradeSnsControlledCanister = record {
+                new_canister_wasm = vec nat8;
+                mode = opt int32;
+                canister_id = opt principal;
+                canister_upgrade_arg = opt vec nat8;
+            };
+        };
+    };
+)'
+```
+
+## `DeregisterDappCanisters` 
+
+### Relevant type signatures
+
+```candid
+    type DeregisterDappCanisters : DeregisterDappCanisters;
+
+    type DeregisterDappCanisters = record {
+        canister_ids : vec principal;
+        new_controllers : vec principal;
+    };
+```
+
+### Putting it together
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "lorem ipsum";
+        url = "lorem ipsum";
+        summary = "lorem ipsum";
+        action = opt variant {
+            DeregisterDappCanisters = record {
+                canister_ids = vec principal;
+                new_controllers = vec principal;
+            };
+        };
+    };
+)'
+```
+
+### Putting it together
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "lorem ipsum";
+        url = "lorem ipsum";
+        summary = "lorem ipsum";
+        action = opt variant {
+            <PROPOSAL_TYPE> = <PARAMETERS_OF_PROPOSAL_TYPE>
+        };
+    }
+)'
+```
+
+## `ManageSnsMetadata` 
+
```

Comment:
```suggestion
Each SNS has a metadata that defines the SNS project and which includes a URL, e.g., under which the main dapp canister can be found, a logo, a name, and a description summarizing the purpose of the projects.
This metadata can be updated at any time, for example if there is a rebranding for the associated project.
To do so, the SNS DAO can use the proposal  `ManageSnsMetadata` which defines all of the above attributes.
```

I would add: for those things that I don't want to change, do I need to leave it empty or do I need to specify all things including those that I don't want to change?